File: solution/array/122-best-time-to-buy-and-sell-stock-ii/solution.js
/**
 * @param {number[]} prices
 * @return {number}
 */
function maxProfit(prices) {
    if (!prices || prices.length < 2) {
        return 0;
    }

    let totalProfit = 0;

    for (let day = 1; day < prices.length; day++) {
        const priceDifference = prices[day] - prices[day - 1];

        if (priceDifference > 0) {
            totalProfit += priceDifference;
        }
    }

    return totalProfit;
}

File: solution/array/122-best-time-to-buy-and-sell-stock-ii/solution.go
package main

func maxProfit(prices []int) int {
	if len(prices) < 2 {
		return 0
	}

	totalProfit := 0

	for day := 1; day < len(prices); day++ {
		priceDifference := prices[day] - prices[day-1]

		if priceDifference > 0 {
			totalProfit += priceDifference
		}
	}

	return totalProfit
}